packages/job-manager/src/batch/pool-load-generator.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { injectable } from 'inversify';
+
+export interface PoolLoad {
+    activeTasks: number;
+    runningTasks: number;
+}
+
+export interface PoolMetricsInfo {
+    id: string;
+    maxTasksPerPool: number;
+    load: PoolLoad;
+}
+
+export interface PoolMetricsState {
+    poolLoad: PoolLoad;
+    tasksIncrementCount: number;
+    processingSpeed: number;
+    activeToRunningTasksRatio: number;
+    samplingIntervalInSeconds: number;
+    timestamp: number;
+}
+
+@injectable()
+export class PoolLoadGenerator {
+    private lastPoolState: PoolMetricsState;
+
+    public get processingSpeedPerMinute(): number {
+        return Math.round((60 / this.lastPoolState.samplingIntervalInSeconds) * this.lastPoolState.processingSpeed);
+    }
+
+    public get samplingIntervalInSeconds(): number {
+        return this.lastPoolState.samplingIntervalInSeconds;
+    }
+
+    public get activeToRunningTasksRatio(): number {
+        return this.lastPoolState.activeToRunningTasksRatio;
+    }
+
+    public setLastTasksIncrementCount(lastTasksIncrementCount: number): void {
+        this.lastPoolState.tasksIncrementCount = lastTasksIncrementCount;
+    }
+
+    public getTasksIncrementCount(poolMetricsInfo: PoolMetricsInfo, targetActiveToRunningTasksRatio: number): number {
```

Comment:
>targetActiveToRunningTasksRatio: number [](start = 68, length = 39)

how about let this class depend on the config instead of passing this every time when this method is called?

Reply:
updated